locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/JcloudsLocation.java

```diff
@@ -2133,11 +2133,17 @@ protected NodeMetadata findNodeOrThrow(ConfigBag config) {
         }
         NodeMetadata node = Iterables.getOnlyElement(candidateNodes);
 
-        String pkd = LocationConfigUtils.getOsCredential(config).checkNoErrors().logAnyWarnings().getPrivateKeyData();
+        OsCredential osCredentials = LocationConfigUtils.getOsCredential(config).checkNoErrors().logAnyWarnings();
+        String pkd = osCredentials.getPrivateKeyData();
+        String password = osCredentials.getPassword();
         if (Strings.isNonBlank(pkd)) {
             LoginCredentials expectedCredentials = LoginCredentials.fromCredentials(new Credentials(user, pkd));
             //override credentials
             node = NodeMetadataBuilder.fromNodeMetadata(node).credentials(expectedCredentials).build();
+        } else if (Strings.isNonBlank(password)) {
+            LoginCredentials expectedCredentials = LoginCredentials.fromCredentials(new Credentials(user, password));
+            //override credentials
+            node = NodeMetadataBuilder.fromNodeMetadata(node).credentials(expectedCredentials).build();
         }
 
         return node;
```

Comment:
I'm going to leave as-is because there is an implicit third case (where neither if block matches). In that case, we don't override the node's credentials (so use whatever the compute-service gave us). I suspect that will not have credentials at all.